tests/util/util.ts

```diff
@@ -19,30 +20,48 @@ export interface KeyPair {
   /** The private key of the keypair, in PEM format */
   privateKeyPem: string;
   /** The public key of the keypair, in JWK format, wrapped as a JWKS */
-  jwks: { keys: [ReturnType<typeof publicKeyToJwk>] };
+  jwks: { keys: Jwk[] };
   /** The public key of the keypair, in JWK format */
-  jwk: ReturnType<typeof publicKeyToJwk>;
-  /** The modulus of the public key of the keypair, as NodeJS buffer */
-  nBuffer: Buffer;
-  /** The exponent of the public key of the keypair, as NodeJS buffer */
-  eBuffer: Buffer;
+  jwk: Jwk;
 }
 
 export function generateKeyPair(
-  deconstructPublicKeyInDerFormat: (publicKey: Buffer) => {
-    n: Buffer;
-    e: Buffer;
-  },
-  options?: { kid?: string; alg?: string }
+  options:
+    | {
+        kty: "RSA";
+        alg?: "RS256" | "RS384" | "RS512";
+        kid?: string;
+        use?: string;
+      }
+    | {
+        kty: "EC";
+        namedCurve: "P-256" | "P-384" | "P-521";
+        alg?: "ES256" | "ES384" | "ES512";
+        kid?: string;
+        use?: string;
+      } = {
+    kty: "RSA",
+    alg: "RS256",
+    use: "sig",
+  }
 ): KeyPair {
-  const { privateKey, publicKey } = generateKeyPairSync("rsa", {
-    modulusLength: 4096,
-    publicExponent: 0x10001,
-  });
-  const jwk = publicKeyToJwk(publicKey, deconstructPublicKeyInDerFormat, {
-    kid: options?.kid,
-    alg: options?.alg,
-  });
+  const { privateKey, publicKey } =
+    options.kty === "RSA"
+      ? generateKeyPairSync("rsa", {
+          modulusLength: 4096,
+          publicExponent: 0x10001,
+        })
+      : generateKeyPairSync("ec", {
+          namedCurve: options.namedCurve,
+        });
+
+  const jwk = publicKey.export({
+    format: "jwk",
+  }) as Jwk;
+  jwk.alg =
+    "alg" in options ? options.alg : options.kty === "RSA" ? "RS256" : "ES256";
```

Comment:
for EC, can we default to ES256 or is that only valid if named curve parameter is "P-256" ?

Reply:
OK